pkg/util/validation.go

```diff
@@ -18,19 +18,34 @@ package util
 
 import (
 	"regexp"
-	"strings"
 )
 
-// IsDNSLabel tests for a string that conforms to the definition of a label in
-// DNS (RFC 1123).
-func IsDNSLabel(value string) bool {
-	return IsDNS1123Label(value)
+const LabelValueFmt string = "[A-Za-z0-9_\\-\\.,]*"
+
+var labelValueRegexp = regexp.MustCompile("^" + LabelValueFmt + "$")
+
+const LabelValueMaxLength int = 63
+
+func IsValidLabelValue(value string) bool {
+	return (len(value) <= LabelValueMaxLength && labelValueRegexp.MatchString(value))
 }
 
-// IsDNSSubdomain tests for a string that conforms to the definition of a
-// subdomain in DNS (RFC 1123).
-func IsDNSSubdomain(value string) bool {
-	return IsDNS1123Subdomain(value)
+const AnnotationValueFmt string = ".*"
+
+var annotationValueRegexp = regexp.MustCompile("^" + AnnotationValueFmt + "$")
+
+const TotalAnnotationSizeB int = 64 * (1 << 10) // 64 kB
+
+func IsValidAnnotationValue(value string) bool {
+	return annotationValueRegexp.MatchString(value)
+}
+
+const QualifiedNameFmt string = "([A-Za-z0-9_\\-\\.]+/)?[A-Za-z0-9_\\-\\.]*[A-Za-z0-9]"
```

Comment:
This I don't understand. This pretty much is the same as previous definition but allowing capital letters. IIUC we wanted to 'add more freedom' in the form of allowing some punctuation in the middle of the 'word'. What about Token := [A-Za-z0-9], ExtendedToken := [-A-Z-a-z0-9_.], NamePiece := (Token ExtendedToken*)?Token, QualifiedName := (NamePiece/)?NamePiece ?